maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/main/java/org/apache/maven/scm/provider/git/jgit/command/branch/JGitBranchCommand.java

```diff
@@ -126,8 +131,12 @@ protected ScmResult executeBranchCommand(
 
             return new BranchScmResult("JGit branch", files);
 
+        } catch (PushException e) {
+            return new BranchScmResult("JGit branch", "Failed to push changes: " + e.getMessage(), "", false);
         } catch (Exception e) {
```

Comment:
can this be a more specific exception?

Reply:
For me this is already as specific as it can get. What do you suggest here?

Reply:
line 136 was not touched in this PR and is not related. This can be done in some other PR!